src/v/kafka/server/connection_context.cc

```diff
@@ -42,6 +42,16 @@ ss::future<> connection_context::process_one_request() {
           if (!sz) {
               return ss::make_ready_future<>();
           }
+          if (sz > _max_request_size()) {
+              vlog(
+                klog.warn,
+                "request from {} is larger ({} bytes) than configured max "
+                "request size {}",
+                _rs.conn->addr,
+                sz,
+                _max_request_size());
+              _rs.conn->shutdown_input();
```

Comment:
I believe you should return a value here to stop execution, maybe `ss::now`

Reply:
This approach will effectively close the connection. After shutting down an input with  ` _rs.conn->shutdown_input()`  all the subsequent reads from the `input_stream` will fail. Next read will happen immediately in the next step either `parse_header` . Throwing an exception would be an option but we would need a special handing for this situation in `net:server` as the exception would be recognized as unhanded exception thrown from service handler and logged with `ERORR` severity which we do not want as this is not an internal redpanda error but rather a client request error.

Reply:
Actually i was testing this further and indeed the `shutdown_input`  may not cause the parse_header to fail in some cases (we need to update the other code paths)

Reply:
@mmaslankaprv is the update to this comment that in this case that control returns back to the event loop and then is_finished_parsing is true and the connection shuts down?

Reply:
exactly, i've change this handling to be exception based as it seems clearer